mantidproject/mantid
# Fix a few 3MF format related issues

**Description of work.**

Fix bug in logging of the number of vertices and triangles loaded where logging strings were being mangled due to absence of std::to_string
Fix bug where couldn't specify an element for the material of an environment component without massdensity
Add a 3mf file describing the PEARL sample and environment to the installer as a compact alternative to the .xml file that names various .stl files

**To test:**

Run CreateSampleWorkspace with default parameters to create a test workspace
Run LoadSampleEnvironment and select the new PearlSampleAndEnvironment.3mf file. This will load a sample shape and mesh shapes for the Paris Edinburgh press sample environment into Mantid
Check it loads OK with the messages like the following written to the log:

![image](https://user-images.githubusercontent.com/56295817/125428625-f539904e-e3a5-47cb-a54d-e479ac774b52.png)


Fixes #31987.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
